### ISSUE TITLE
Respell functions output wrong octave in some cases

### ISSUE THREAD

**Issue by gilbertohasnofb** [Abjad/abjad#1249]
The functions `abjad.Accidental.respell_with_flats()` and `abjad.Accidental.respell_with_sharp()` output pitches in the wrong octaves in some cases. These are cases when a B# needs to be respelled as C or when a Cb needs to be respelled as a B. Below is one example:

```
>>> staff = abjad.Staff(r"gs'4 as'4 bs'4 cs''4")
>>> abjad.f(staff)
\new Staff
{
    gs'4
    as'4
    bs'4
    cs''4
}
```
![image](https://user-images.githubusercontent.com/5700795/101820212-30e05d00-3b1e-11eb-87cb-6a9a41b23603.png)

```
>>> staff = abjad.Staff(r"gs'4 as'4 bs'4 cs''4")
>>> abjad.Accidental.respell_with_flats(staff)
>>> abjad.f(staff)
\new Staff
{
    af'4
    bf'4
    c'4
    df''4
}
```

![image](https://user-images.githubusercontent.com/5700795/101820259-3b025b80-3b1e-11eb-91e9-00e94bb5a564.png)

edit: tested in Abjad 3.1, but issue seems to be present in the current `master` branch

**Comment by gilbertohasnofb** [comment]
The issue seems to be the way that  `abjad.Pitch._respell_with_flats()` and `abjad.Pitch._respell_with_sharps()` handle ocatves. They keep the same value of `self.octave.number` when respelling pitch, but in the case of B# -> C or Cb -> B that number changes:

```
>>> note = abjad.Note("bs'")
>>> note.written_pitch.pitch_class.number
0
>>> note.written_pitch.octave.number
4
>>> 
>>> note = abjad.Note("c''")
>>> note.written_pitch.pitch_class.number
0
>>> note.written_pitch.octave.number
5
```

**Comment by trevorbaca** [comment]
Confirmed that the bug persists in 3.2:
```
staff = abjad.Staff(r"gs'4 as'4 bs'4 cs''4")
abjad.iterpitches.respell_with_flats(staff)
abjad.f(staff)
\new Staff
{
    af'4
    bf'4
    c'4
    df''4
}
```

Note also that I've moved the respell functions to a new `abjad.iterpitches` module in 3.2

**Comment by trevorbaca** [comment]
@gilbertohasnofb 

Done, and merged into master.
```
staff = abjad.Staff(r"gs'4 as'4 bs'4 cs''4")
abjad.iterpitches.respell_with_flats(staff)
abjad.f(staff)
\new Staff
{
    af'4
    bf'4
    c''4
    df''4
}
```
<img width="196" alt="respell" src="https://user-images.githubusercontent.com/5804733/101833132-84df4780-3b06-11eb-9929-cb1b10c39632.png">

Thanks for the report.

Will be included in 3.2!

**Comment by gilbertohasnofb** [comment]
@trevorbaca Thanks for the quick fix, Trevor! Just tried it out and it works perfectly
